tests_util/src/db.rs

```diff
@@ -0,0 +1,79 @@
+//! In memory databse which simulates disk database reads
+//! 
+//! According to this article, LevelDB can perform 129.000 random read operations / s
+//! http://www.lmdb.tech/bench/microbench/benchmark.html
+
+use std::{time, thread};
+use std::cell::RefCell;
+use std::collections::{HashMap, HashSet};
+use bloomchain::{Position, Bloom, BloomDatabase};
+use bloomchain::group::{GroupPosition, BloomGroup, BloomGroupDatabase};
+
+/// 1.0s / 129 000
+const LEVELDB_READ_TIME: u32 = 7752;
+
+#[derive(Default)]
+pub struct BloomMemoryDatabase {
+	/// Read time in nanoseconds
+	read_time: u32,
+	mem: HashMap<Position, Bloom>,
+	queried: RefCell<HashSet<Position>>,
```

Comment:
You can avoid having two separate structures and lookups and use:
```rust
type Accessed = bool;
mem: HashMap<Position, (Bloom, Cell<Accessed>)>,
```
not entirely sure how this will affect the benchmarks though.